genesis_core/user_api/iam/dm/models.py

```diff
@@ -1029,6 +1154,38 @@ class IamClient(
         ra_types.Url(),
         required=True,
     )
+    rules = properties.property(
+        ra_types.TypedList(AbstractValidationRule.get_rule_selector()),
+        default=list,
+        required=False,
+    )
+
+    def get_validation_rules(self):
+        """Returns normalized validation rule objects."""
+        if not self.rules:
+            return []
+        
+        rules = []
+        for rule_model in self.rules:
+            kind = rule_model.kind
+            cache = type(self)._verifier_cache
+            if kind not in cache:
+                cache[kind] = u.load_from_entry_point(ENTRY_POINT_GROUP, kind)
+            verifier_cls = cache[kind]
```

Comment:
Minor performance thing (this might be a hot method, so its kinda relevant): 
maybe do it like this
```
verifier_cls = u.load_from_entry_point(ENTRY_POINT_GROUP, kind)
cache[kind] = verifier_cls
rules.append(ValidationRule(rule_model, verifier_cls))
```
instead of a dict lookup (`verifier_cls = cache[kind]`) right after inserting.